File: logging/logging-annotation-processor/src/test/java/io/koraframework/logging/aspect/mdc/MDCContextHolder.java
package io.koraframework.logging.aspect.mdc;

import io.koraframework.logging.common.arg.StructuredArgumentWriter;

import java.util.Collections;
import java.util.Map;

public class MDCContextHolder {

    private Map<String, StructuredArgumentWriter> mdcContext;

    public Map<String, StructuredArgumentWriter> get() {
        return mdcContext;
    }

    public void set(Map<String, StructuredArgumentWriter> mdcContext) {
        this.mdcContext = mdcContext == null
            ? Collections.emptyMap()
            : Map.copyOf(mdcContext);
    }
}

File: logging/logging-aop/logging-aop-symbol-processor/src/test/kotlin/io/koraframework/logging/symbol/processor/aop/mdc/MDCContextHolder.kt
package io.koraframework.logging.symbol.processor.aop.mdc

import io.koraframework.logging.common.arg.StructuredArgumentWriter

class MDCContextHolder {

    private var mdcContext: Map<String, StructuredArgumentWriter>? = null

    fun get(): Map<String, StructuredArgumentWriter>? {
        return mdcContext
    }

    fun set(mdcContext: Map<String, StructuredArgumentWriter>?) {
        this.mdcContext = mdcContext?.toMap() ?: emptyMap()
    }
}
